### PR TITLE
`ultralytics 8.3.210` Skip non-finite training forward passes

### PR DESCRIPTION
**Pull request by glenn-jocher**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Improves training stability by skipping non-finite loss batches and bumps version to 8.3.210 🚀

### 📊 Key Changes
- Version updated to 8.3.210 in `ultralytics/__init__.py` 🧩
- Trainer now checks for non-finite losses (NaN/Inf) during the forward pass and skips those batches with a warning: “Non-finite forward pass, skipping batch...” 🛡️
- Minor tweak to running training loss (`tloss`) calculation to initialize cleanly when first set

### 🎯 Purpose & Impact
- Prevents crashes and unstable updates by not backpropagating through invalid losses, especially helpful with mixed precision training ⚖️
- Improves robustness when encountering problematic data or extreme hyperparameters (e.g., high learning rates) ✅
- Users may see warnings and slightly fewer processed batches per epoch, but overall training should be more reliable and consistent 📈

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! This is an automated response to help streamline the review process. An Ultralytics engineer will also review and assist shortly. In the meantime, please double-check the following:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

Since this PR addresses non-finite losses during training, if this change is in response to a bug, please include a Minimum Reproducible Example (MRE) to help us validate and test the behavior end-to-end:
- Environment: OS, Python, PyTorch, CUDA
- Ultralytics version: commit SHA or `pip show ultralytics`
- Exact command used, e.g.:
  - `yolo train model=yolo11n.pt data=coco8.yaml imgsz=640 epochs=3 device=0`
- Minimal config, dataset sample, and any custom loss/augmentation code that triggers NaN/Inf
- Logs showing the non-finite loss occurrence and frequency

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🛡️✨

**Comment by glenn-jocher** [comment]
@Laughing-q add this step to detect nonfinite forward pass

**Comment by UltralyticsAssistant** [comment]
Fantastic work, @glenn-jocher — PR [#22353](https://github.com/ultralytics/ultralytics/pull/22353) is merged! 🚀  
“Quality is not an act, it is a habit.” — Aristotle. Your update embodies this: skipping non-finite loss batches, cleaner `tloss` init, and the 8.3.210 version bump together make training more resilient, especially under mixed precision and aggressive hyperparameters. This is the kind of polish that keeps our users confident and productive. 🙌

To everyone training models: update to 8.3.210 for improved stability: `pip install -U ultralytics`
